pallets/liquid-staking-two/src/lib.rs

```diff
@@ -0,0 +1,820 @@
+    /// For stake request, after all xtoken have been mint, this may take 6 hours in Kusama, and one day in Polkadot
+    /// The data can be removed after user claim their xToken
+    /// the corresponding era's operation in `StakingOperationHistory` should also be successed.
+    ///
+    /// For unstake request, xtoken need to be burn, and get token back,
+    /// the data can be removed after the lock period passed and user claim.
+    /// this may take 7 days in Kusama, and 28 days in Polkadot.
+    /// the corresponding era's operation in `StakingOperationHistory` should also be successed.
+    #[pallet::storage]
+    #[pallet::getter(fn matching_queue)]
+    pub type MatchingQueue<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        T::AccountId,
+        Blake2_128Concat,
+        EraIndex,
+        MatchingUserBuffer,
+        ValueQuery,
+    >;
+
+    /// Previous era index on Parachain.
+    ///
+    /// Considering that in case stake client cannot update era index one by one.
+    #[pallet::storage]
+    #[pallet::getter(fn previous_era)]
+    pub type PreviousEra<T: Config> = StorageValue<_, EraIndex, ValueQuery>;
+
+    /// Current era index on Relaychain.
+    ///
+    /// CurrentEra: EraIndex
+    #[pallet::storage]
+    #[pallet::getter(fn current_era)]
+    pub type CurrentEra<T: Config> = StorageValue<_, EraIndex, ValueQuery>;
+
+    /// Store operations and corresponding status during each era
+    ///
+    /// The operation include: bond/bond_extra/unbond/rebond/on_new_era/transfer_to_relaychain/record_reward/record_slash
+    /// The status include: start/processing/successed/failed
+    #[pallet::storage]
+    #[pallet::getter(fn staking_operation_history)]
+    pub type StakingOperationHistory<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        EraIndex,
+        Blake2_128Concat,
+        StakingOperationType,
+        Operation<T::BlockNumber>,
+    >;
+
+    // #[pallet::genesis_config]
+    // pub struct GenesisConfig {
+    //     pub exchange_rate: Rate,
+    //     pub reserve_factor: Ratio,
+    // }
+
+    // #[cfg(feature = "std")]
+    // impl Default for GenesisConfig {
+    //     fn default() -> Self {
+    //         Self {
+    //             exchange_rate: Rate::default(),
+    //             reserve_factor: Ratio::default(),
+    //         }
+    //     }
+    // }
+
+    // #[pallet::genesis_build]
+    // impl<T: Config> GenesisBuild<T> for GenesisConfig {
+    //     fn build(&self) {
+    //         ExchangeRate::<T>::put(self.exchange_rate);
+    //         ReserveFactor::<T>::put(self.reserve_factor);
+    //     }
+    // }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T>
+    where
+        [u8; 32]: From<<T as frame_system::Config>::AccountId>,
+    {
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn trigger_new_era(
```

Comment:
we should use this function only to update `CurrentEra` or `PreviousEra`, another function is needed to handle matching result and do real operations

Reply:
Yes, we have another function.